src/ui/components/snackBar/SnackBar.tsx

```diff
@@ -0,0 +1,140 @@
+  visible: boolean
+  message?: string
+  icon?: FunctionComponent<IconInterface>
+  onClose?: () => void
+  timeout?: number
+  backgroundColor: ColorsEnum
+  color: ColorsEnum
+  animationDuration?: number
+}
+
+const _SnackBar = (props: SnackBarProps) => {
+  const Icon = props.icon
+  const animationDuration = props.animationDuration || 500
+
+  const containerRef: RefType = useRef(null)
+  const [isVisible, setVisible] = useState(false)
+
+  const triggerVanishAnimation = useCallback(
+    async () =>
+      containerRef?.current?.fadeOutUp(animationDuration).then(() => void setVisible(false)),
+    []
+  )
+  const triggerApparitionAnimation = useCallback(async () => {
+    setVisible(true)
+    containerRef?.current?.fadeInDown(animationDuration)
+  }, [])
+
+  const onClose = useCallback((e: GestureResponderEvent) => {
+    e.stopPropagation()
+    props.onClose?.()
+  }, [])
+
+  // Visibility effect
+  useEffect(() => {
+    const shouldDisplay = props.visible && !isVisible
+    const shouldHide = !props.visible && isVisible
+
+    if (shouldDisplay) {
+      triggerApparitionAnimation()
+    }
+    if (shouldHide) {
+      triggerVanishAnimation()
+    }
+    // Timeout section
+    if (!props.timeout || !props.onClose || shouldHide) {
+      return
+    }
+    const timeout = setTimeout(props.onClose, props.timeout)
+    return () => clearTimeout(timeout)
+  }, [props.visible])
+
+  const { top } = useSafeAreaInsets()
+
+  return (
+    <AnimatedContainer
+      backgroundColor={props.backgroundColor}
+      marginTop={top}
+      easing="ease"
+      duration={animationDuration}
+      ref={containerRef}>
+      <SnackBarContainer isVisible={isVisible}>
+        <React.Fragment>
+          <ContentContainer testID="toasterContainer">
+            {Icon && <Icon size={22} color={props.color} />}
+            <Text color={props.color}>{props.message}</Text>
+          </ContentContainer>
+          <CloseIconContainer onPress={onClose}>
+            <Close size={24} color={props.color} />
+          </CloseIconContainer>
+        </React.Fragment>
+      </SnackBarContainer>
+    </AnimatedContainer>
+  )
+}
+
+export const SnackBar = memo(_SnackBar)
+
+const AnimatedContainer = styled(AnimatableView)<{ backgroundColor: string; marginTop: number }>(
+  ({ marginTop = 0 }) => ({
+    position: 'absolute',
+    marginTop,
+    top: 0,
+    left: 0,
+    right: 0,
+    zIndex: ZIndexes.SNACK_BAR,
+  })
+)
+
+const SnackBarContainer = styled.View<{ isVisible: boolean }>(({ isVisible }) => ({
+  flexDirection: 'row',
+  alignItems: 'center',
+  padding: getSpacing(2),
+  flexGrow: 0,
+  display: isVisible ? 'flex' : 'none',
+}))
+
+const ContentContainer = styled.View({
+  flex: 1,
+  flexDirection: 'row',
+  alignItems: 'center',
+  marginLeft: getSpacing(1),
+})
+
+const Text = styled.Text<{ color: string }>(({ color }) => ({
+  color,
+  marginLeft: getSpacing(1),
+  flexGrow: 0,
+  maxWidth: Dimensions.get('window').width - 80,
```

Comment:
Une alternative ? Je me dis que sur tablette ca peut devenir relou d'avoir une gigantesque snaclbar.
```suggestion
  maxWidth: getSpacing(125),
```

Reply:
est-ce que `getSpacing(125)` est dynamique ?? moi je veux une taille limitée pas une taille fixe

Reply:
Mon point était de mettre une max width fixe, mais c'est peut être pas le comportement voulu.